e2e/kind.sh

```diff
@@ -175,18 +168,27 @@ test_locationmesh_peer() {
 test_fullmesh() {
 	# shellcheck disable=SC2016
 	$KUBECTL_BINARY patch ds -n kube-system kilo -p '{"spec": {"template":{"spec":{"containers":[{"name":"kilo","args":["--hostname=$(NODE_NAME)","--create-interface=false","--kubeconfig=/etc/kubernetes/kubeconfig","--mesh-granularity=full"]}]}}}}'
-	sleep 5
 	block_until_ready_by_name kube-system kilo-userspace 
 	$KUBECTL_BINARY wait pod -l app.kubernetes.io/name=adjacency --for=condition=Ready --timeout 3m
-	retry 30 5 "" check_ping
-	sleep 5
-	retry 10 5 "the adjacency matrix is not complete yet" check_adjacent
+	assert "retry 30 5 '' check_ping" "should be able to ping all Pods"
+	assert "retry 10 5 'the adjacency matrix is not complete yet' check_adjacent 12" "adjacency should return the right number of successful pings"
+	echo "sleep for 30s (one reconciliation period) and try again..."
+	sleep 30
+	assert "retry 10 5 'the adjacency matrix is not complete yet' check_adjacent 12" "adjacency should return the right number of successful pings after reconciling"
 }
 
 test_fullmesh_peer() {
```

Comment:
At least in the former version of bash_unit, the order of the test function was not defined by their position in the file. I think we need to patch the kilo manifest.

Reply:
Officially bash_unit provides no guarantee about the order in which tests are run. In practice, they are executed in lexicographic order. This works today, but I think that the future might be to split this out into multiple test files where each file has it's own `setup_suite`/`teardown_suite` functions. This way we could ensure everything in one file has, for example, a working `full-mesh` cluster. What do you think?

Reply:
Yes sounds very good to invoke bash unit twice with different setup suits.